### ISSUE TITLE
Adding logging

### ISSUE THREAD

**Issue by robzajac** [zixuanzh/py-libp2p#84]
It would be nice for py-libp2p to do proper logging for network events and other noteworthy events through a standard Python logging framework.

**Comment by AliabbasMerchant** [comment]
Shall I work on this??  
I could use the python `logging` module   
@robzajac  @zixuanzh

**Comment by zixuanzh** [comment]
@AliabbasMerchant yes, that will be great! We are working with Python 3.6 and above. Thanks!

**Comment by zaibon** [comment]
I've been also looking around for nice logging library to use for this project, and I stumble upon: https://www.structlog.org/

I kinda like this approach of structured logging, specially for network library like this one, I found the result pretty nice and easy to read. It is also 100% compatible with logging from standard python library.

**Comment by AliabbasMerchant** [comment]
I have restarted work on this.  
I am hoping to complete it in 2-3 days.  
A few doubts:  
1. Do we wanna use `structlog` or the standard python logger??
2. I have an idea of making a decorator, especially for logging.
Sample:
```
@log(log_level, name)
def my_func(a, b=1):
    return a+b
ans = my_func(5,b=3)  # ans = 8
```
Output:
```
TIME: NAME: LOG_LEVEL: my_func(5,b=3) called
TIME: NAME: LOG_LEVEL: my_func(5,b=3) returned 8
```
Additionally, I will make a logging function, having a similar functionality, so that it can be called wherever needed.

Tagging: @robzajac @zixuanzh

**Comment by robzajac** [comment]
Hi @AliabbasMerchant, we'd like to conform to the [logging style used in py-evm](https://github.com/ethereum/py-evm/blob/master/eth/tools/logging.py) for consistency. This means we'd like to use the standard Python logging framework - feel free to draw additional inspiration from py-evm.

CC @ralexstokes
